rs/nns/governance/src/governance.rs

```diff
@@ -6139,7 +6139,15 @@ impl Governance {
         // Get the balance of the neuron's subaccount from ledger canister.
         let account = neuron_subaccount(subaccount);
         tla_log_locals! { account: tla::account_to_tla(account), neuron_id: nid.id };
-        let balance = self.ledger.account_balance(account).await?;
+        let balance = self
+            .ledger
+            .account_balance(account)
+            .await
+            .inspect_err(|_| {
+                // remove neuron cannot fail here, as we have already added the neuron.
+                let _ = self.remove_neuron(neuron.clone());
```

Comment:
If remove_neuron returns Err, let's make that "bubble up" to the caller of claim_neuron.

> That should never happen.
> --Famous Last Words

> Errors should never pass silently.
> --Zen of Python

I think you are implicitly assuming that there are no bugs? Ideally, we should avoid such assumptions. Ofc, it is impractical to assume that bugs are totally PERVASIVE, but that's not what I'm saying. Rather, if it is not "very" difficult, let's try to limit the spread of "fires" by building more "fire walls".